### PR TITLE
TSK -726 Adapt taskana history plugin to Taskana core interface

### PR DESCRIPTION
**Pull request by MiguelMartinRojas**

### CODE REVIEW

File: lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/ExampleBootstrap.java [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
+        this.historyService.initialize(taskanaEngineConfiguration);
+    }
 
     @PostConstruct
-    public void test() {
-        HistoryEvent newEvent = historyService.newHistoryEvent("", "", "");
-        historyService.createHistoryEvent(newEvent);
-        HistoryEvent insertedEvent = historyService.getHistoryEvent(newEvent.getId());
-        System.out.println("---------------------------> event inserted: " + insertedEvent.getId());
+    public void testBootstraping() {
```
Comment (holgerhagen) [inline lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/ExampleBootstrap.java]: If this is a test, why don't we create an acceptance test instead? 

I understand, you've just refactored that, but wouldn't that be better?

File: lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/TaskanaHistoryPluginConfiguration.java [hunk 2]
```diff
@@ -10,8 +10,9 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Primary;
 
-import pro.taskana.history.plugin.configuration.TaskanaHistoryEngineConfiguration;
-import pro.taskana.history.plugin.impl.TaskanaHistoryEngineImpl;
+import pro.taskana.configuration.TaskanaEngineConfiguration;
+import pro.taskana.history.api.TaskanaHistory;
+import pro.taskana.simplehistory.impl.HistoryEventServiceImpl;
 
 public class TaskanaHistoryPluginConfiguration {
```
Comment (holgerhagen) [inline lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/TaskanaHistoryPluginConfiguration.java]: We don't call it "Plugin" anymore, do we? Maybe you can change this to the new naming scheme? E.g. "TaskanaSimpleHistoryProviderConfiguration"?

File: lib/taskana-simplehistory-provider/pom.xml [hunk 3]
```diff
@@ -166,6 +166,11 @@
 			<artifactId>mybatis</artifactId>
 			<version>3.4.6</version>
 		</dependency>
+		<dependency>
+			<groupId>pro.taskana</groupId>
+			<artifactId>taskana-core</artifactId>
+			<version>1.0.4-SNAPSHOT</version>
```
Comment (holgerhagen) [inline lib/taskana-simplehistory-provider/pom.xml]: We should not depent on snapshots, right?

File: lib/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/HistoryEventServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import pro.taskana.configuration.TaskanaEngineConfiguration;
+import pro.taskana.history.api.TaskanaHistory;
+import pro.taskana.history.api.TaskanaHistoryEvent;
+import pro.taskana.impl.util.LoggerUtils;
+import pro.taskana.simplehistory.impl.mappings.HistoryEventMapper;
+
+/**
+ * This is the implementation of HistoryService.
+ */
+public class HistoryEventServiceImpl implements TaskanaHistory {
```
Comment (holgerhagen) [inline lib/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/HistoryEventServiceImpl.java]: Why did you rename it to HistoryEVENTService? Is there a reason for it?

File: lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/ExampleBootstrap.java (hunk 1 above)
Comment (MiguelMartinRojas) [reply lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/ExampleBootstrap.java]: I just followed the same structure and naming as taskana-spring-example. This is not a test it is just a bootstrap example.  I will change the bootstraping example, I haven't done tests because right now the logic is covered by this bootstraping. I think that we can add tests after adding more functionality.

File: lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/TaskanaHistoryPluginConfiguration.java (hunk 2 above)
Comment (MiguelMartinRojas) [reply lib/taskana-simplehistory-example/src/main/java/pro/taskana/simplehistory/TaskanaHistoryPluginConfiguration.java]: Agree

File: lib/taskana-simplehistory-provider/pom.xml (hunk 3 above)
Comment (MiguelMartinRojas) [reply lib/taskana-simplehistory-provider/pom.xml]: Yes. I will change it.

File: lib/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/HistoryEventServiceImpl.java (hunk 4 above)
Comment (MiguelMartinRojas) [reply lib/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/HistoryEventServiceImpl.java]: I Thought that HistoryEventService is a name closer to the funtionality that we are going to implement.
